advent-of-code/src/main/java/aoc/year2019/day06/UniversalOrbitMap.java

```diff
@@ -0,0 +1,207 @@
+            return directOrbits;
+        }
+    }
+
+    /**
+     * Count total number of direct and indirect orbits between space objects given in the file data
+     */
+    private void countTotalNumberOfOrbits() {
+        // Number of orbits (direct and indirect) per space object
+        Map<String, Integer> numberOfOrbit = new HashMap<>();
+
+        // Counting each orbit for each space object in orbit
+        this.directOrbits.forEach(directOrbit -> {
+            if (null == directOrbit) {
+                return;
+            }
+
+            int count = this.countNumberOfOrbitsForOneSpaceObject(directOrbit);
+            numberOfOrbit.put(directOrbit.objectInOrbit, count);
+        });
+
+        // Sum of all the orbits and display of result
+        int totalOrbits = numberOfOrbit.values().stream().reduce(0, Integer::sum);
+        log.info("Total number of orbits : {}", totalOrbits);
+    }
+
+    /**
+     * Count number of direct and indirect orbits for one space object
+     *
+     * @param firstDirectOrbit Direct orbit of the space object in orbit
+     * @return Number of orbits linked to the first space object in orbit
+     */
+    private int countNumberOfOrbitsForOneSpaceObject(DirectOrbit firstDirectOrbit) {
+        int count = 1; // By default, every space object in orbit has 1 orbit
+
+        DirectOrbit currentDirectOrbit = firstDirectOrbit;
+
+        while (null != currentDirectOrbit) {
+            String objectInOrbit = currentDirectOrbit.center;
+            currentDirectOrbit = this.directOrbits.stream()
+                    .filter(filteredDirectOrbit -> Objects.equals(filteredDirectOrbit.objectInOrbit, objectInOrbit))
+                    .findFirst()
+                    .orElse(null);
+            if (null != currentDirectOrbit) {
+                count += 1;
+            }
+        }
+
+        return count;
+    }
+
+    /**
+     * Calculate how many orbital transfers it takes to go from one space object to another
+     *
+     * @param source      Name of space object we use to start our transfers
+     * @param destination Name of space object we use as our destination
+     */
+    private void calculateOrbitalTransfers(String source, String destination) {
+        // Find paths of source and destination to COM (universal Center Of Mass)
+        List<String> sourceToCom = this.findPathToCOM(source);
+        List<String> destinationToCom = this.findPathToCOM(destination);
+
+        // Find step where paths intersect
+        String commonStep = destinationToCom.stream()
+                .filter(sourceToCom::contains)
+                .findFirst()
+                .orElse(null);
+
+        // If a common step is found, we remove it and the following ones
+        if (null != commonStep) {
+            this.cleanDuplicates(sourceToCom, commonStep);
+            this.cleanDuplicates(destinationToCom, commonStep);
+        } else {
+            throw new NoSuchElementException();
+        }
```

Comment:
A bas le culte du `null` ! 🔥 🪕 
```suggestion
                .orElseThrow();

        // If a common step is found, we remove it and the following ones
        this.cleanDuplicates(sourceToCom, commonStep);
        this.cleanDuplicates(destinationToCom, commonStep);
```